README.md

```diff
@@ -1,37 +1,69 @@
 
-The UID 2 Project is subject to Tech Lab IPR’s Policy and is managed by the IAB Tech Lab Addressability Working Group and Privacy & Rearc Commit Group. Please review the governance rules [here](https://github.com/IABTechLab/uid2-core/blob/master/Software%20Development%20and%20Release%20Procedures.md)
+The UID 2 Project is subject to Tech Lab IPR’s Policy and is managed by the IAB Tech Lab Addressability Working Group and Privacy & Rearc Commit Group. Please review the governance rules [here](https://github.com/IABTechLab/uid2-core/blob/master/Software%20Development%20and%20Release%20Procedures.md).
 
-## Build & Package
+# Requirements
 
-```
-    mvn package
-```
+* Java 1.8 or later
 
-## Install
 
-1. add reference to uid2-client-2.0.0-jar-with-dependencies.jar generated from build & package setup in your project
+## Installation
 
-2. install the library locally and reference via maven
+Add this dependency to your project's POM:
 
-```
-    mvn install:install-file -Dfile="./target/uid2-client-2.0.0-jar-with-dependencies.jar" -DgroupId="com.uid2" -DartifactId="uid2-client" -Dpackaging=jar -Dversion="2.0.0"
-```
-
-this will install the jar into your local .m2 repository, you can set your desired target repo with `-DlocalRepositoryPath=path-to-specific-local-repo`
-
-then on your maven project's pom.xml, add
 
 ```
         <dependency>
             <groupId>com.uid2</groupId>
             <artifactId>uid2-client</artifactId>
-            <version>2.0.0</version>
+            <version>4.0.0</version>
         </dependency>
 ```
 
-## Examples
+## Usage for DSPs
+
+See com.uid2.client.test.IntegrationExamples for example usage for DSPs.
+
+## Usage for Publishers
+You can use this SDK with your own HTTP client library, as outlined in the steps below. An example application is at [uid2-examples](https://github.com/UnifiedID2/uid2-examples/tree/main/publisher).
+
+
+1. Create an instance of PublisherUid2Helper as an instance variable. 
+
+    `private final PublisherUid2Helper publisherUid2Helper = new PublisherUid2Helper(UID2_SECRET_KEY);`
+1. When a user authenticates and authorizes the creation of a UID2:
+
+    `Envelope envelope = publisherUid2Helper.createEnvelope(IdentityInput.fromEmail(emailAddress));`
+1. Using an HTTP client library of your choice, post this envelope to the [/v2/token/generate](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md) endpoint, with:
+   1. Header: `Authorization: Bearer {UID2_API_KEY}`
+   1. Body: `envelope.getEnvelope()`
+1. If the HTTP response status code is 200:
```

Comment:
What should they do if it is not 200?

Reply:
changed this line to:
If the HTTP response status code is _not_ 200, see [Response Status Codes](https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md#response-status-codes) to determine next steps. Otherwise: